src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -230,7 +230,7 @@ export class DataViewsService {
    * @param force
    */
   setDefault = async (id: string | null, force = false) => {
-    if (force || !this.config.get('defaultIndex')) {
+    if (force || !(await this.config.get('defaultIndex'))) {
```

Comment:
Looks like we had a logic bug

Not sure how that functional test worked before, but it relies on `default` index pattern and `default` was missing because we didn't set `default` during index pattern creation